### ISSUE TITLE
Negative loss

### ISSUE THREAD

**Issue by wh1t3tea** [FrontierLabs/F5R-TTS#10]
During the first stage of training, after ~70k steps the loss becomes negative:

<img width="523" height="364" alt="Image" src="https://github.com/user-attachments/assets/55ae348e-2c47-4a3e-acd0-eb97741bb869" />

The model doesn’t collapse, but the negative loss still confuses me. Is it okay?

**Comment by Sakkov** [comment]
Yes, that's okay. It's due to how the GRPO signal is introduced to the loss.
```
loss = - pro_advantages + loss_kl
```

**Comment by wh1t3tea** [comment]
> Yes, that's okay. It's due to how the GRPO signal is introduced to the loss.
> 
> ```
> loss = - pro_advantages + loss_kl
> ```

I was talking about first stage. I removed (t * t) * ln_sig part from loss and it looks better. However model still can’t coverage well, especially on samples longer than 10 seconds. I guess the problem might be that it hacks loss function minimising ln_sigma rather than flow - mu
